### DISCUSSION TITLE
IapExample: Products.tsx vs ClassSetup.tsx

### DISCUSSION THREAD

**Discussion by PatriciaSauer** [hyochan/react-native-iap#2791] [Q&A]
Hi,

I'm still trying to understand how this library works. So I investigated the example app a lot. 

I still don't understand why in Product.tsx `finishTransaction` is wrapped in `if ((isIosStorekit2() && currentPurchase?.transactionId) || currentPurchase?.transactionReceipt)` (see [here](https://github.com/dooboolab-community/react-native-iap/blob/main/IapExample/src/screens/Products.tsx#L59)) whereas in ClassSetup.tsx we are not wrapping in such an if (see [here](https://github.com/dooboolab-community/react-native-iap/blob/main/IapExample/src/screens/ClassSetup.tsx#L96)). 

Is the one in Products.tsx for iOS and the other one for Android? Why do I need both implementations? What is the dependency on `currentPurchase` of the `useEffect` hook for? Can't I just implement everything in the purchaseUpdateListener?

Unfortunately, I was not able to run the example app. Maybe someone is able to provide detailed steps to run this app.

Any help would be appreciated. Thanks in advance

**Comment by brent-charlton** [comment]
I believe the dependency of currentPurchase in the use effect  is just an alternative way to respond to the purchase update listener when using hooks. It's not clear to me either whether it's best to use hooks or follow the guidance in the docs about avoiding async await
